cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -779,11 +924,9 @@ impl ABIBody for AArch64ABIBody {
                     }
                     _ => {}
                 };
-                ret.push(store_stack(
-                    MemArg::FPOffset(self.fp_to_arg_offset() + off),
-                    from_reg.to_reg(),
-                    ty,
-                ))
+                let mem = MemArg::reg_maybe_offset(self.ret_area_ptr.unwrap().to_reg(), off, ty)
+                    .expect("Return-value area is too large");
```

Comment:
Since this is basically user-controlled, I think this could happen in practice, right? If so, could we bubble up the error with an ImplLimit error, or open an issue to not forget about it?

Reply:
So, this sent me on a bit of an adventure to add a new `RegOffset` pseudo-amode that does the Right Thing during emission (e.g. synthesize a large offset with a constant and an add). This had some carryover effects elsewhere so I'll PTAL the patch back to you before merging :-)